pkgs/build-support/go/module.nix

```diff
@@ -149,10 +153,16 @@ let
   package = stdenv.mkDerivation (args // {
     nativeBuildInputs = [ go ] ++ nativeBuildInputs;
 
-    inherit (go) GOOS GOARCH;
+    env = (args.env or { }) // goPlatform.env.host // {
+      inherit CGO_ENABLED GO111MODULE GOTOOLCHAIN;
+      GOFLAGS = lib.concatStringsSep " " (
```

Comment:
Why is this being changed to use `concatStringsSep` and `lib.optional`?

Reply:
`concatStringsSep` because `env` does not accept lists.
https://github.com/NixOS/nixpkgs/blob/74970765262f0958d124b5b43afdd45e8285d527/pkgs/stdenv/generic/make-derivation.nix#L490-L491

`lib.optional` — no particular reason, I’ll change it back to `lib.optionals`, I think I’ve changed that accidentally in one place and then though that it was `lib.optional` all along 🥲